cross-dom-bridge/index.js

```diff
@@ -0,0 +1,233 @@
+
+  console.log(`On L1:${l1Balance} Gwei    On L2:${l2Balance} Gwei`)
+}    // reportBalances
+
+
+
+const reportERC20Balances = async () => {
+  const l1Balance = (await l1ERC20.balanceOf(addr)).toString().slice(0,-18)
+  const l2Balance = (await l2ERC20.balanceOf(addr)).toString().slice(0,-18)
+  console.log(`DAI on L1:${l1Balance}     DAI on L2:${l2Balance}`)  
+}    // reportERC20Balances
+
+const depositETH = async () => {
+
+  console.log("Deposit ETH")
+  await reportBalances()
+  const start = new Date()
+
+  const response = await crossChainMessenger.depositETH(gwei)
+  console.log(`Transaction hash (on L1): ${response.hash}`)
+  await response.wait()
+  console.log("Waiting for status to change to RELAYED")
+  console.log(`Time so far ${(new Date()-start)/1000} seconds`)
+  await crossChainMessenger.waitForMessageStatus(response.hash, 
+                                                  optimismSDK.MessageStatus.RELAYED, 
+                                                  {pollIntervalMs: 1000})
+
+  await reportBalances()    
+  console.log(`depositETH took ${(new Date()-start)/1000} seconds\n\n`)
+}     // depositETH()
+
+
+
+const depositERC20 = async () => {
+
+  console.log("Deposit ERC20")
+  await reportERC20Balances()
+  const start = new Date()
+
+  // Need the l2 address to know which bridge is responsible
+  const allowanceResponse = await crossChainMessenger.approveERC20(
+    daiAddrs.l1Addr, daiAddrs.l2Addr, dai)
+  await allowanceResponse.wait()
+  console.log(`Allowance given by tx ${allowanceResponse.hash}`)
+  console.log(`Time so far ${(new Date()-start)/1000} seconds`)
+
+  const response = await crossChainMessenger.depositERC20(
+    daiAddrs.l1Addr, daiAddrs.l2Addr, dai)
+  console.log(`Deposit transaction hash (on L1): ${response.hash}`)
+  await response.wait()
+  console.log("Waiting for status to change to RELAYED")
+  console.log(`Time so far ${(new Date()-start)/1000} seconds`)  
+  await crossChainMessenger.waitForMessageStatus(response.hash, 
+                                                  optimismSDK.MessageStatus.RELAYED, 
+                                                  {pollIntervalMs: 1000})
+
+  await reportERC20Balances()    
+  console.log(`depositERC20 took ${(new Date()-start)/1000} seconds\n\n`)
+}     // depositETH()
+
+
+
+const withdrawERC20 = async () => { 
+  
+  console.log("Withdraw ERC20")
+  const start = new Date()  
+  await reportERC20Balances()
+
+  const response = await crossChainMessenger.withdrawERC20(
+    daiAddrs.l1Addr, daiAddrs.l2Addr, dai)
+  console.log(`Transaction hash (on L2): ${response.hash}`)
+  await response.wait()
+
+  console.log("Waiting for status to change to IN_CHALLENGE_PERIOD")
+  console.log(`Time so far ${(new Date()-start)/1000} seconds`)
+  await crossChainMessenger.waitForMessageStatus(response.hash, 
+    optimismSDK.MessageStatus.IN_CHALLENGE_PERIOD, 
+    {pollIntervalMs: 1000}) 
```

Comment:
I would recommend either not including `pollIntervalMs` here or including a comment that explains what this is doing and gives a reasonable value for people to use in production. 1s is too fast for production use-cases (the default is 4s, which is more reasonable).

Reply:
I had an error at some point because `pollIntervalMs` was unspecified. Now that it's fixed, I'll remove the timeouts.

Reply:
This works:

```js
  await crossChainMessenger.waitForMessageStatus(response.hash, 
                                                  optimismSDK.MessageStatus.RELAYED, {}) 
```

This, however fails:
```js
  await crossChainMessenger.waitForMessageStatus(response.hash, 
                                                  optimismSDK.MessageStatus.RELAYED) 
```

With the error:
```
TypeError: Cannot read property 'timeoutMs' of undefined
    at CrossChainMessenger.waitForMessageStatus (/Users/ori/data/mergers/optimism-tutorial/cross-dom-bridge/node_modules/@eth-optimism/sdk/dist/cross-chain-messenger.js:354:36)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async depositETH (/Users/ori/data/mergers/optimism-tutorial/cross-dom-bridge/index.js:106:3)
    at async main (/Users/ori/data/mergers/optimism-tutorial/cross-dom-bridge/index.js:207:5)

```

I don't know TypeScript. Can you make `opts` optional?

```ts
  public async waitForMessageStatus(
    message: MessageLike,
    status: MessageStatus,
    opts: {
      pollIntervalMs?: number
      timeoutMs?: number
    } = {}
  ): Promise<void> {
```

Reply:
SDK version [0.2.5](https://www.npmjs.com/package/@eth-optimism/sdk/v/0.2.5) should fix this.